src/main/kotlin/gg/essential/elementa/components/Window.kt

```diff
@@ -357,13 +357,20 @@ class Window @JvmOverloads constructor(
             )
         }
 
-        fun ofOrNull(component: UIComponent): Window? {
+        fun ofOrNull(component: UIComponent): Window? = component.cachedWindow ?: run {
             var current = component
+            var isHidden = false
 
-            while (current !is Window && current.hasParent && current.parent != current)
+            while (current !is Window && current.hasParent && current.parent != current) {
+                isHidden = isHidden || current !in current.parent.children
                 current = current.parent
+            }
 
-            return current as? Window
+            (current as? Window).also {
+                if (!isHidden) {
+                    component.cachedWindow = it
```

Comment:
We don't need to **ever** set this I believe. When the component is hidden, we mustn't set it because we can't guarantee that it'll be updated. And when the component is not hidden, it should already have been set via the recursive update, no?

Reply:
Oh, right, yeah. Not sure why I didn't realise that...